### PR TITLE
Add Collector security documentation

### PR DESCRIPTION
**Pull request by tiffany76**
This PR moves **end user** security documentation from a [README](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/security-best-practices.md) in the Collector core repository to the OTel docs website. 

Based on decisions in previous issues and PRs, the following are assumed:
- Security documentation intended for component developers will remain in the README, which will be amended in a follow-up PR.
- Both documents will cross-reference the other.
- The top-level Collector landing page and the Collector configuration page will link to the security documents.

Tracking issue: #3479
Related to: #3227

**NOTE:** Much of the work for this PR was done by @mjingle in #3652. With her permission, I am building on her efforts.

### CODE REVIEW

**Comment by tiffany76** [comment]
Still lots to do. I'll let everyone know when it's in a state fit for reviewing. :+1:

File: content/en/docs/security/config-best-practices.md
Comment (reyang) [inline content/en/docs/security/config-best-practices.md:15]: "Ports" seem to be too specific, consider rephrasing it to "exposes a minimum attack surface" - it could be TCP/UDP ports, component dependency (SBOM), etc.

File: content/en/docs/security/config-best-practices.md
Comment (reyang) [inline content/en/docs/security/config-best-practices.md:14]: "one receiver and one exporter" seems very strict, what if the user wants to fork the data to two different targets? (e.g. are we suggesting that forking data is not recommended from security perspective?)

File: content/en/docs/security/config-best-practices.md
Comment (reyang) [inline content/en/docs/security/config-best-practices.md:25]: Resource exhaustion is a great topic, I feel the current description is a bit misleading - even if the settings are correct, OpenTelemetry Collector can still cause resource exhaustion - is this the current state or there is already a solid solution for resource governance?

File: content/en/docs/security/config-best-practices.md
Comment (reyang) [inline content/en/docs/security/config-best-practices.md:41]: I don't recall the exact details, would this bind to both IPv4 and IPv6, or the Collector has a recommendation regarding how to specify IPv4/v6 explicitly?

File: content/en/docs/security/config-best-practices.md
Comment (jpkrohling) [inline content/en/docs/security/config-best-practices.md:49]: We are in the future already!

File: content/en/docs/security/config-best-practices.md
Comment (jpkrohling) [inline content/en/docs/security/config-best-practices.md:66]: I don't think we should prescribe which processors people should use. We could provide some example configurations for common use-cases though.

File: content/en/docs/security/config-best-practices.md
Comment (jpkrohling) [inline content/en/docs/security/config-best-practices.md:94]: Given the discussions we are having around the batch processor, we should probably stop advertising it.

cc @mx-psi

Comment (jpkrohling) [reply content/en/docs/security/config-best-practices.md:94]: After a second thought, perhaps we should stop talking about the memory limiter _processor_ as well, mentioning the _extension_ instead once it gets ready?

https://github.com/open-telemetry/opentelemetry-collector/tree/main/extension/memorylimiterextension

File: content/en/docs/security/config-best-practices.md
Comment (jpkrohling) [inline content/en/docs/security/config-best-practices.md:123]: I feel like this is misplaced here... I don't think the forwarding extension is related to security, nor is a best practice.

File: content/en/docs/security/config-best-practices.md
Comment (jpkrohling) [inline content/en/docs/security/config-best-practices.md:147]: It's OK -- what we shouldn't be telling people is to run things as root or blindly disable security protections (like selinux).

File: content/en/docs/security/config-best-practices.md
Comment (jpkrohling) [inline content/en/docs/security/config-best-practices.md:152]: I'm also not sure where this is coming from: the only component that I know of spawning new processes is the jmx receiver.

File: content/en/docs/security/hosting-best-practices.md
Comment (jpkrohling) [inline content/en/docs/security/hosting-best-practices.md:13]: Perhaps add a reference to the secrets management practices for Kubernetes?

File: content/en/docs/security/hosting-best-practices.md
Comment (jpkrohling) [inline content/en/docs/security/hosting-best-practices.md:28]:
```suggestion
- Enable authentication, for instance by using bearer token authentication extensions and basic
```

File: content/en/docs/security/config-best-practices.md
Comment (tiffany76) [reply content/en/docs/security/config-best-practices.md:14]: Thanks, all! The content so far has been copied from an earlier draft of this documentation. I think this sentence was an edit of the following statement in the core respository's [README](https://github.com/open-telemetry/opentelemetry-collector/blob/d6f568dd5b6ddfd87a7c6350545cb818f902a06f/docs/security-best-practices.md?plain=1#L63):

> The configuration drives the Collector's behavior and care should be taken to ensure the configuration only enables the minimum set of capabilities and as such exposes the minimum set of required ports.

I will remove the "one receiver and one exporter" restriction from this PR.

File: content/en/docs/security/_index.md
Comment (tiffany76) [reply content/en/docs/security/_index.md:14]: Still searching for the existing guidance, but I've updated the wording as suggested. I'll keep looking for the PII reference.

File: content/en/docs/security/config-best-practices.md
Comment (tiffany76) [reply content/en/docs/security/config-best-practices.md]: I've reworked this whole section (Receivers and exporters) based on the comments. Please let me know what you think. No rush. I still have lots to do. Thanks!

File: content/en/docs/security/config-best-practices.md
Comment (tiffany76) [reply content/en/docs/security/config-best-practices.md]: Hi @reyang, I removed the mention of resource exhaustion here. The topic is addressed in more detail in the Processors section. I'll let you know once I start work on it. Thanks!

File: content/en/docs/security/config-best-practices.md
Comment (tiffany76) [reply content/en/docs/security/config-best-practices.md]: Changed in cbf67a9. Thanks!

File: content/en/docs/security/config-best-practices.md
Comment (tiffany76) [reply content/en/docs/security/config-best-practices.md]: I've rewritten the DOS safeguard section as well, with updated information about `localhost`. 

Rather than provide an example of binding to a pod's IP, I linked to the [README section](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/security-best-practices.md#safeguards-against-denial-of-service-attacks). Unless you think that content (Docker, Kubernetes, etc.) is relevant to end users and belongs on the website? I was trying to draw a line between end user and developer content, but I'm happy to make changes.

File: content/en/docs/security/config-best-practices.md
Comment (tiffany76) [reply content/en/docs/security/config-best-practices.md:125]: Based on @reyang's earlier comment that resource utilization is an important topic, it seems like we should say something about best practices here. If we don't recommend the batch and memory limiter processors, and the memory limiter extension is still in development, how should we advise users to safeguard resource utilization in their Collector configurations and hosting?

File: content/en/docs/security/config-best-practices.md
Comment (tiffany76) [reply content/en/docs/security/config-best-practices.md:183]: This section comes directly from the [security README](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/security-best-practices.md#subprocesses). Should I remove it?

File: content/en/docs/security/config-best-practices.md
Comment (tiffany76) [reply content/en/docs/security/config-best-practices.md]: Removed in dbfd1b6. Thanks!

File: content/en/docs/security/config-best-practices.md
Comment (tiffany76) [reply content/en/docs/security/config-best-practices.md]: Added a recommendation to use `ocb` and clarified that unused components should be removed from configurations. Thanks!

File: content/en/docs/security/config-best-practices.md
Comment (tiffany76) [reply content/en/docs/security/config-best-practices.md]: Hi again. I updated the resource utilization section in 8605abb.

File: content/en/docs/security/config-best-practices.md
Comment (tiffany76) [reply content/en/docs/security/config-best-practices.md]: I took a shot at writing this section on my own (8605abb), but it could definitely use a developer's review. 

I mention batching and memory limiting, but I don't call out the processors or extensions specifically. Once the memory limiter extension is ready, we can update with an example.

File: content/en/docs/security/config-best-practices.md
Comment (tiffany76) [reply content/en/docs/security/config-best-practices.md]: Removed the section in favor of examples.

File: content/en/docs/security/config-best-practices.md
Comment (mx-psi) [inline content/en/docs/security/config-best-practices.md:55]: I don't think the push-pull thing matters much here. There are also other components you may care about properly securing (e.g. if you are using an https confmap provider or an extension):
```suggestion
- Receivers, exporters and other components should establish
  network connections over a secure channel, potentially
  authenticated as well.
```

File: content/en/docs/security/config-best-practices.md
Comment (mx-psi) [inline content/en/docs/security/config-best-practices.md:80]: The `4317` and `4318` is specific to the OTLP receiver, not all components.
```suggestion
From Collector v0.110.0, the default host for all servers in Collector components is `localhost`. 
For earlier versions of the Collector, change the default endpoint
from `0.0.0.0` to `localhost` in all components by enabling the
`component.UseLocalHostAsDefaultHost`
[feature gate](https://github.com/open-telemetry/opentelemetry-collector/tree/main/featuregate).
```

File: content/en/docs/security/config-best-practices.md
Comment (mx-psi) [inline content/en/docs/security/config-best-practices.md:205]: This is mentioned above already in a different fashion, I would just remove this section

File: content/en/docs/security/config-best-practices.md
Comment (mx-psi) [inline content/en/docs/security/config-best-practices.md:213]: I would remove this for now and add it back later (we can create an issue for this). I don't think we necessarily need this for the first version

File: content/en/docs/security/config-best-practices.md
Comment (mx-psi) [reply content/en/docs/security/config-best-practices.md:231]: I vote for removing this section

File: content/en/docs/security/_index.md
Comment (tiffany76) [reply content/en/docs/security/_index.md:19]: The trouble is that the links to the Collector security pages don't exist yet, until this PR is merged.

File: content/en/docs/security/config-best-practices.md
Comment (tiffany76) [reply content/en/docs/security/config-best-practices.md:47]: Sounds good to me.

File: content/en/docs/security/config-best-practices.md
Comment (tiffany76) [reply content/en/docs/security/config-best-practices.md]: Removed in 961ac47. I also created #5429 so we remember to iterate.

File: content/en/docs/security/config-best-practices.md
Comment (tiffany76) [reply content/en/docs/security/config-best-practices.md]: Removed in 961ac47.

File: content/en/docs/security/config-best-practices.md
Comment (tiffany76) [reply content/en/docs/security/config-best-practices.md]: Removed in 961ac47.

File: content/en/docs/security/hosting-best-practices.md
Comment (tiffany76) [reply content/en/docs/security/hosting-best-practices.md]: I added this in 73f4c38.

**Comment by tiffany76** [comment]
Hi folks! I'm moving this PR out of draft. It's ready for review (finally)!

Just a couple thoughts:

- I think I have addressed all the comments so far, but I did rewrite most of the content, so you might have trouble finding your comments in the changed files. I'm sorry about that.
- I just reread @jpkrohling's [comment](https://github.com/open-telemetry/opentelemetry.io/pull/3652#issuecomment-1853643945) about keeping these docs closer to the Collector getting started docs. I know there was some back and forth about it. I don't have a preference either way and will move the files to the Collector docs, if that's what everyone agrees on.
- I will be AFK next week, so that will be a prime time for everyone to pile on the reviews. :smiley:

File: content/en/docs/security/_index.md
Comment (mx-psi) [reply content/en/docs/security/_index.md:19]: Ah, alright!

File: content/en/docs/security/config-best-practices.md
Comment (mx-psi) [reply content/en/docs/security/config-best-practices.md]: I think a general mention to batching and memory limiting should be more future-proof

File: content/en/docs/security/hosting-best-practices.md
Comment (mx-psi) [inline content/en/docs/security/hosting-best-practices.md:48]: I wouldn't use "scrape" here, since the metrics may be exported via OTLP (scrape makes it sound like we only support Prometheus for this, which is not really the case). Maybe just:

```suggestion
performance. Collect metrics from the Collector about its CPU, memory, and
```

File: content/en/docs/security/_index.md
Comment (svrnm) [inline content/en/docs/security/_index.md:8]: should we move that into the frontmatter `description`, it reads like one

**Review by svrnm — APPROVED** [review]
minor comment, but looks good to me overall, great work! 🎉

File: content/en/docs/security/hosting-best-practices.md
Comment (mx-psi) [inline content/en/docs/security/hosting-best-practices.md:33]: This section is worded differently from the others, I would rephrase it so that it more explicitly tells you what to do (use the least amount of privilege required).

File: content/en/docs/security/hosting-best-practices.md
Comment (mx-psi) [inline content/en/docs/security/hosting-best-practices.md:19]: To avoid the repetition maybe we can say this instead:
```suggestion
## Keep your secrets safe
```

File: content/en/docs/security/hosting-best-practices.md
Comment (tiffany76) [reply content/en/docs/security/hosting-best-practices.md:33]: Good catch on the change in wording! I used the example that @jpkrohling gave in a comment that I can't find now. I will work on rephrasing it to match the other examples on the page.

File: content/en/docs/security/_index.md
Comment (tiffany76) [reply content/en/docs/security/_index.md:8]: It might be a little long for the `description`. Let me take another shot at making the wording more intro-like.

Comment (tiffany76) [reply content/en/docs/security/_index.md]: I rewrote this in f099457.

File: content/en/docs/security/hosting-best-practices.md
Comment (tiffany76) [reply content/en/docs/security/hosting-best-practices.md]: I rewrote the section in 3b86e29. Let me know what you think. Thanks!

**Comment by tiffany76** [comment]
@open-telemetry/sig-security-maintainers, @open-telemetry/collector-approvers, @open-telemetry/docs-approvers: Hi all, I'm asking for a last round of reviews. Please let me know if you'd like me to make any changes.

Also, please see my earlier comment and add your thoughts:

> I just reread @jpkrohling's [comment](https://github.com/open-telemetry/opentelemetry.io/pull/3652#issuecomment-1853643945) about keeping these docs closer to the Collector getting started docs. I know there was some back and forth about it. I don't have a preference either way and will move the files to the Collector docs, if that's what everyone agrees on.

**Review by mx-psi — APPROVED** [review]
This looks good to me for a first version

File: content/en/docs/security/hosting-best-practices.md
Comment (reyang) [inline content/en/docs/security/hosting-best-practices.md:43]:
```suggestion
servers. To limit access to authorized users, you should:
```

**Review by reyang — APPROVED** [review]
LGTM, well written!

**Comment by tiffany76** [comment]
We have an approval from each SIG, so I was thinking we could give it one more day and then merge this on Friday. How does that sound? 

@jpkrohling @open-telemetry/docs-approvers

**Comment by reyang** [comment]
> We have an approval from each SIG, so I was thinking we could give it one more day and then merge this on Friday. How does that sound?
> 
> @jpkrohling @open-telemetry/docs-approvers

:shipit: 🚢

**Review by chalin — APPROVED** [review]
Overall looks good (did a quick scan, not an in depth read though).
I've added suggested fixes for section-local paths.
I'll let other maintainers give their formal approval for the content.

**Review by cartermp — APPROVED** [review]

**Pull request merged**
